src/main/java/org/testng/TestNG.java

```diff
@@ -211,11 +210,10 @@ private void init(boolean useDefaultListeners) {
   }
 
   public int getStatus() {
-    return m_status;
-  }
-
-  private void setStatus(int status) {
-    m_status |= status;
+    if (m_hasTests) {
```

Comment:
I prefer:

```java
if (!m_hasTests) {
   return HAS_NO_TEST;
}
return exitCode.getExitCode();
```

because getStatus return the exit code, except in a specific case.

Reply:
Now that I have started reading `Clean Code` by `Robert Martin`, I can understand what you are saying. Intent.. its the intent that matters :) Fixed.